src/main/java/Entities/CurrentUser.java

```diff
@@ -0,0 +1,26 @@
+package Entities;
+import Entities.User;
+
+/**
+ * Singleton Class of the current user of the application.
+ */
+public class CurrentUser {
+    private static CurrentUser currentUser;
+    private static User user;
+
+    private CurrentUser(){}
+    public static CurrentUser getInstance(){
+
+        if (currentUser == null){
+            currentUser = new CurrentUser();
+        }
+        return currentUser;
+    }
+    public static void setUser(User newUser){
+        user = newUser;
+    }
+    public static User getUser(){
```

Comment:
this makes sense to me! nice implementation -- I think you've followed the design pattern exactly!